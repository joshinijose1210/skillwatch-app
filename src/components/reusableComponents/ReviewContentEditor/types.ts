export type Props = {
    id: string | number;
    action: string;
    richTextValue: string;
    richTextError: string;
    feedbackTag?: number;
    handleInputChange: (data: { id: string | number; name: string; value: any }) => void;
    ratingOptions: { label: string; value: any }[];
    rating: any;
    ratingError: string;
    ratingLabel?: string;
    ratingPlaceholder?: string;
    useBlur?: boolean;
    reviewPlaceholder?: string;
    isOptional?: boolean;
};
