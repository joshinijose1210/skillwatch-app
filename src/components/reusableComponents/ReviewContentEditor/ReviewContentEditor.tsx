import { RatingPicker, RichTextEditorWrapper } from '@pages/SelfReview/SelfReviewForm/SelfReviewForm.styled';
import { FC, memo, useCallback } from 'react';
import { RichTextEditor } from '../RichTextEditor';
import { Props } from './types';

export const ReviewContentEditor: FC<Props> = memo(
    ({
        id,
        action,
        richTextValue,
        handleInputChange,
        ratingOptions,
        rating,
        ratingError,
        feedbackTag,
        ratingLabel,
        reviewPlaceholder,
        ratingPlaceholder,
        useBlur,
        isOptional = false
    }) => {
        const handleOnChange = useCallback(
            (value: number | string | boolean) => {
                handleInputChange({ id, name: 'rating', value });
            },
            [id, handleInputChange]
        );

        const handleChange = useCallback(
            (value: string) => {
                handleInputChange({ id, name: 'response', value });
            },
            [handleInputChange, id]
        );

        return (
            <>
                <RichTextEditorWrapper>
                    <RichTextEditor
                        fullWidth
                        height={150}
                        bgColor="#fff"
                        placeholder={reviewPlaceholder}
                        disabled={action === 'View'}
                        editor={'feedback'}
                        input={richTextValue}
                        marginBottom={1}
                        setInput={handleChange}
                        feedbackTag={feedbackTag}
                        useBlur={useBlur}
                        isOptional={isOptional}
                    />
                </RichTextEditorWrapper>
                <RatingPicker
                    id="rating"
                    variant="outlined"
                    options={ratingOptions}
                    label={ratingLabel}
                    placeholder={ratingPlaceholder}
                    value={rating}
                    size="M"
                    minWidth="100%"
                    disabled={action === 'View'}
                    errorText={ratingError}
                    onChange={handleOnChange}
                    data-testid={`kpi-${id}`}
                />
            </>
        );
    }
);

ReviewContentEditor.displayName = 'ReviewContentEditor';
