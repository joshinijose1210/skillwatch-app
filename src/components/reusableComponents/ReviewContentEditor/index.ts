export { ReviewContentEditor as default } from './ReviewContentEditor';
