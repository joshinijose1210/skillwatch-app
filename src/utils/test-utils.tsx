import { ToastContainer } from '@medly-components/core';
import { setupListeners } from '@reduxjs/toolkit/dist/query';
import { store as rtkStore, setupStore } from '@slice';
import { RenderOptions, RenderResult, render } from '@testing-library/react';
import { defaultTheme } from '@theme';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import reduxMockStore from 'redux-mock-store';
import { ThemeProvider } from 'styled-components';
import { userMockState } from '../mocks/userMockState';
import { initialState } from '../slice';

interface CustomOptions extends RenderOptions {
    initialState?: any;
    initialEntries?: any;
}

interface WithThemeProviderProps {
    children?: React.ReactNode;
}

const mockStore = reduxMockStore(),
    store = mockStore({
        ...initialState,
        user: { ...initialState.user, groups: ['admin'] }
    });

const WithThemeProvider: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <ThemeProvider theme={defaultTheme}>
        <>
            <ToastContainer position="top-end" />
            {children}
        </>
    </ThemeProvider>
);

const WithStore: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <WithThemeProvider>{children}</WithThemeProvider>
    </Provider>
);

const WithRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <MemoryRouter>
        <WithThemeProvider>{children}</WithThemeProvider>
    </MemoryRouter>
);

const WithStoreAndRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <MemoryRouter>
            <WithThemeProvider>{children}</WithThemeProvider>
        </MemoryRouter>
    </Provider>
);

export const RTKRender = (ui: React.ReactElement, { initialState = userMockState, ...renderOptions }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);

    const RtkWrapper: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
        <Provider store={store}>
            <MemoryRouter initialEntries={renderOptions.initialEntries}>
                <WithThemeProvider>{children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
    return { rtkStore, ...render(ui, { wrapper: RtkWrapper, ...renderOptions }) };
};
const customRender = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithThemeProvider, ...options });

export const renderWithStore = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStore, ...options });

export const renderWithRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithRouter, ...options });

export const renderWithStoreAndRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStoreAndRouter, ...options });

export const wrapper = (props: { children: React.ReactNode }, { initialState = userMockState }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);
    return (
        <Provider store={store}>
            <MemoryRouter>
                <WithThemeProvider>{props.children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render, store };
