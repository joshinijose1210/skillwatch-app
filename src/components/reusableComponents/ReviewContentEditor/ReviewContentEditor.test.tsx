import { RTKRender } from '@utils/test-utils';
import { ReviewContentEditor } from './ReviewContentEditor';

describe('ReviewContentEditor', () => {
    it('should render properly', () => {
        const { container } = RTKRender(
            <ReviewContentEditor
                id={''}
                action={''}
                richTextValue={''}
                richTextError={''}
                handleInputChange={jest.fn()}
                ratingOptions={[]}
                rating={undefined}
                ratingError={''}
            />
        );
        expect(container).toMatchSnapshot();
    });
});
