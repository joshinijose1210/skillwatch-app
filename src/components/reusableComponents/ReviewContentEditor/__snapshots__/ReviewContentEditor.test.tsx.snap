// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewContentEditor should render properly 1`] = `
<div>
  <div
    class="sc-zzdsm5-17 jDDkji"
  >
    <div
      class="sc-1wh81lg-1 ctCvwI"
      height="150"
      marginbottom="1"
    >
      <div
        bgcolor="#fff"
        class="sc-1wh81lg-0 brgywW"
        editor="feedback"
      >
        <div
          class="ql-toolbar ql-snow"
        >
          <span
            class="ql-formats"
          >
            <button
              class="ql-bold"
              type="button"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <path
                  class="ql-stroke"
                  d="M5,4H9.5A2.5,2.5,0,0,1,12,6.5v0A2.5,2.5,0,0,1,9.5,9H5A0,0,0,0,1,5,9V4A0,0,0,0,1,5,4Z"
                />
                 
                <path
                  class="ql-stroke"
                  d="M5,9h5.5A2.5,2.5,0,0,1,13,11.5v0A2.5,2.5,0,0,1,10.5,14H5a0,0,0,0,1,0,0V9A0,0,0,0,1,5,9Z"
                />
                 
              </svg>
            </button>
            <button
              class="ql-italic"
              type="button"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <line
                  class="ql-stroke"
                  x1="7"
                  x2="13"
                  y1="4"
                  y2="4"
                />
                 
                <line
                  class="ql-stroke"
                  x1="5"
                  x2="11"
                  y1="14"
                  y2="14"
                />
                 
                <line
                  class="ql-stroke"
                  x1="8"
                  x2="10"
                  y1="14"
                  y2="4"
                />
                 
              </svg>
            </button>
            <button
              class="ql-underline"
              type="button"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <path
                  class="ql-stroke"
                  d="M5,3V9a4.012,4.012,0,0,0,4,4H9a4.012,4.012,0,0,0,4-4V3"
                />
                 
                <rect
                  class="ql-fill"
                  height="1"
                  rx="0.5"
                  ry="0.5"
                  width="12"
                  x="3"
                  y="15"
                />
                 
              </svg>
            </button>
            <button
              class="ql-strike"
              type="button"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <line
                  class="ql-stroke ql-thin"
                  x1="15.5"
                  x2="2.5"
                  y1="8.5"
                  y2="9.5"
                />
                 
                <path
                  class="ql-fill"
                  d="M9.007,8C6.542,7.791,6,7.519,6,6.5,6,5.792,7.283,5,9,5c1.571,0,2.765.679,2.969,1.309a1,1,0,0,0,1.9-.617C13.356,4.106,11.354,3,9,3,6.2,3,4,4.538,4,6.5a3.2,3.2,0,0,0,.5,1.843Z"
                />
                 
                <path
                  class="ql-fill"
                  d="M8.984,10C11.457,10.208,12,10.479,12,11.5c0,0.708-1.283,1.5-3,1.5-1.571,0-2.765-.679-2.969-1.309a1,1,0,1,0-1.9.617C4.644,13.894,6.646,15,9,15c2.8,0,5-1.538,5-3.5a3.2,3.2,0,0,0-.5-1.843Z"
                />
                 
              </svg>
            </button>
          </span>
          <span
            class="ql-formats"
          >
            <span
              class="ql-align ql-picker ql-icon-picker"
            >
              <span
                aria-controls="ql-picker-options-0"
                aria-expanded="false"
                class="ql-picker-label"
                role="button"
                tabindex="0"
              >
                <svg
                  viewBox="0 0 18 18"
                >
                   
                  <line
                    class="ql-stroke"
                    x1="3"
                    x2="15"
                    y1="9"
                    y2="9"
                  />
                   
                  <line
                    class="ql-stroke"
                    x1="3"
                    x2="13"
                    y1="14"
                    y2="14"
                  />
                   
                  <line
                    class="ql-stroke"
                    x1="3"
                    x2="9"
                    y1="4"
                    y2="4"
                  />
                   
                </svg>
              </span>
              <span
                aria-hidden="true"
                class="ql-picker-options"
                id="ql-picker-options-0"
                tabindex="-1"
              >
                <span
                  class="ql-picker-item ql-selected"
                  role="button"
                  tabindex="0"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="3"
                      x2="15"
                      y1="9"
                      y2="9"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="3"
                      x2="13"
                      y1="14"
                      y2="14"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="3"
                      x2="9"
                      y1="4"
                      y2="4"
                    />
                     
                  </svg>
                </span>
                <span
                  class="ql-picker-item"
                  data-value="center"
                  role="button"
                  tabindex="0"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="15"
                      x2="3"
                      y1="9"
                      y2="9"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="14"
                      x2="4"
                      y1="14"
                      y2="14"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="12"
                      x2="6"
                      y1="4"
                      y2="4"
                    />
                     
                  </svg>
                </span>
                <span
                  class="ql-picker-item"
                  data-value="right"
                  role="button"
                  tabindex="0"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="15"
                      x2="3"
                      y1="9"
                      y2="9"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="15"
                      x2="5"
                      y1="14"
                      y2="14"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="15"
                      x2="9"
                      y1="4"
                      y2="4"
                    />
                     
                  </svg>
                </span>
                <span
                  class="ql-picker-item"
                  data-value="justify"
                  role="button"
                  tabindex="0"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="15"
                      x2="3"
                      y1="9"
                      y2="9"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="15"
                      x2="3"
                      y1="14"
                      y2="14"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="15"
                      x2="3"
                      y1="4"
                      y2="4"
                    />
                     
                  </svg>
                </span>
              </span>
            </span>
            <select
              class="ql-align"
              style="display: none;"
            >
              <option
                selected="selected"
              />
              <option
                value="center"
              />
              <option
                value="right"
              />
              <option
                value="justify"
              />
            </select>
          </span>
          <span
            class="ql-formats"
          >
            <button
              class="ql-list"
              type="button"
              value="ordered"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <line
                  class="ql-stroke"
                  x1="7"
                  x2="15"
                  y1="4"
                  y2="4"
                />
                 
                <line
                  class="ql-stroke"
                  x1="7"
                  x2="15"
                  y1="9"
                  y2="9"
                />
                 
                <line
                  class="ql-stroke"
                  x1="7"
                  x2="15"
                  y1="14"
                  y2="14"
                />
                 
                <line
                  class="ql-stroke ql-thin"
                  x1="2.5"
                  x2="4.5"
                  y1="5.5"
                  y2="5.5"
                />
                 
                <path
                  class="ql-fill"
                  d="M3.5,6A0.5,0.5,0,0,1,3,5.5V3.085l-0.276.138A0.5,0.5,0,0,1,2.053,3c-0.124-.247-0.023-0.324.224-0.447l1-.5A0.5,0.5,0,0,1,4,2.5v3A0.5,0.5,0,0,1,3.5,6Z"
                />
                 
                <path
                  class="ql-stroke ql-thin"
                  d="M4.5,10.5h-2c0-.234,1.85-1.076,1.85-2.234A0.959,0.959,0,0,0,2.5,8.156"
                />
                 
                <path
                  class="ql-stroke ql-thin"
                  d="M2.5,14.846a0.959,0.959,0,0,0,1.85-.109A0.7,0.7,0,0,0,3.75,14a0.688,0.688,0,0,0,.6-0.736,0.959,0.959,0,0,0-1.85-.109"
                />
                 
              </svg>
            </button>
            <button
              class="ql-list"
              type="button"
              value="bullet"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <line
                  class="ql-stroke"
                  x1="6"
                  x2="15"
                  y1="4"
                  y2="4"
                />
                 
                <line
                  class="ql-stroke"
                  x1="6"
                  x2="15"
                  y1="9"
                  y2="9"
                />
                 
                <line
                  class="ql-stroke"
                  x1="6"
                  x2="15"
                  y1="14"
                  y2="14"
                />
                 
                <line
                  class="ql-stroke"
                  x1="3"
                  x2="3"
                  y1="4"
                  y2="4"
                />
                 
                <line
                  class="ql-stroke"
                  x1="3"
                  x2="3"
                  y1="9"
                  y2="9"
                />
                 
                <line
                  class="ql-stroke"
                  x1="3"
                  x2="3"
                  y1="14"
                  y2="14"
                />
                 
              </svg>
            </button>
          </span>
          <span
            class="ql-formats"
          >
            <button
              class="ql-link"
              type="button"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <line
                  class="ql-stroke"
                  x1="7"
                  x2="11"
                  y1="7"
                  y2="11"
                />
                 
                <path
                  class="ql-even ql-stroke"
                  d="M8.9,4.577a3.476,3.476,0,0,1,.36,4.679A3.476,3.476,0,0,1,4.577,8.9C3.185,7.5,2.035,6.4,4.217,4.217S7.5,3.185,8.9,4.577Z"
                />
                 
                <path
                  class="ql-even ql-stroke"
                  d="M13.423,9.1a3.476,3.476,0,0,0-4.679-.36,3.476,3.476,0,0,0,.36,4.679c1.392,1.392,2.5,2.542,4.679.36S14.815,10.5,13.423,9.1Z"
                />
                 
              </svg>
            </button>
          </span>
        </div>
        <div
          bgcolor="#fff"
          class="sc-1wh81lg-2 gIrMBs ql-container ql-snow"
          data-testid="textquill"
          height="150"
        >
          <div
            class="ql-editor ql-blank"
            contenteditable="true"
            data-gramm="false"
            data-placeholder="Write your feedback here..."
          >
            <p>
              <br />
            </p>
          </div>
          <div
            class="ql-clipboard"
            contenteditable="true"
            tabindex="-1"
          />
          <div
            class="ql-tooltip ql-hidden"
          >
            <a
              class="ql-preview"
              href="about:blank"
              rel="noopener noreferrer"
              target="_blank"
            />
            <input
              data-formula="e=mc^2"
              data-link="https://quilljs.com"
              data-video="Embed URL"
              type="text"
            />
            <a
              class="ql-action"
            />
            <a
              class="ql-remove"
            />
          </div>
        </div>
        <div
          class="sc-1wh81lg-3 dQvbgb"
        >
          0/1000
        </div>
        <span
          class="sc-1ntgl3l-0 fbvXZC"
          textalign="initial"
          textcolor="red"
          textvariant="body2"
        />
      </div>
    </div>
  </div>
  <div
    class="sc-1hcsztp-0 fjPkSb sc-zzdsm5-6 fueHsi"
    id="rating-wrapper"
    minwidth="100%"
    variant="outlined"
  >
    <div
      class="sc-17d4s39-0 khebHA"
      id="rating-input-wrapper"
      minwidth="100%"
    >
      <div
        class="sc-19px6zp-0 BrSGZ"
        filled="[object Object]"
        fusion="[object Object]"
        minrows="1"
        outlined="[object Object]"
        textvariant="[object Object]"
        variant="outlined"
      >
        <div
          class="sc-lhr232-0 btNOKw"
          variant="outlined"
        >
          <input
            aria-describedby="rating-helper-text"
            autocomplete="off"
            class="sc-hb99wz-0 eESZta"
            data-testid="kpi-"
            errortext=""
            id="rating-input"
            inputmode="none"
            inputsize="M"
            placeholder="Please Select . . ."
            type="text"
            value=""
            variant="outlined"
          />
          <label
            class="sc-1du51u0-0 eMZrST"
            for="rating-input"
            inputwidth="0"
            variant="outlined"
          />
        </div>
        <svg
          class="sc-1pf3x8a-0 klDMsE sc-dynlvq-0 bdmviR"
          fill="none"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
            fill="#000"
          />
        </svg>
      </div>
      
    </div>
  </div>
</div>
`;
